ibis/backends/sql/compilers/clickhouse.py

```diff
@@ -604,6 +601,27 @@ def visit_ArrayUnion(self, op, *, left, right):
     def visit_ArrayZip(self, op: ops.ArrayZip, *, arg, **_: Any) -> str:
         return self.f.arrayZip(*arg)
 
+    def visit_ArrayCollect(self, op, *, arg, where, order_by, ignore_null):
+        if not ignore_null:
+            raise com.UnsupportedOperationError(
+                "`ignore_null=False` is not supported by the clickhouse backend"
```

Comment:
The duplication of this code everywhere is a bit annoying. This _could_ be done in a generic rewrite rule (that doesn't rewrite but just errors if the `ignore_null` is False), but honestly that'd only save a few lines of code (there's only a few backends which do support these aggregations but don't support this option) and increase coupling - duplicating seemed cleaner IMO.